iam/role/role.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/acorn-io/services/aws/libs/common"
+	"github.com/aws/aws-cdk-go/awscdk/v2"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awsiam"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/constructs-go/constructs/v10"
+	"github.com/aws/jsii-runtime-go"
+	"github.com/sirupsen/logrus"
+)
+
+type IAMRoleStackProps struct {
+	StackProps                awscdk.StackProps
+	Tags                      map[string]string      `json:"tags"`
+	RoleName                  string                 `json:"roleName"`
+	TrustedArn                string                 `json:"trustedArn"`
+	Policy                    map[string]interface{} `json:"policy"`
+	MaxSessionDurationMinutes int                    `json:"maxSessionDurationMinutes"`
+	Path                      string                 `json:"path"`
+	ExternalIds               string                 `json:"externalIds"`
+}
+
+func (rsp *IAMRoleStackProps) setDefaults() {
+	if rsp.RoleName == "" {
+		rsp.RoleName = os.Getenv("ACORN_EXTERNAL_ID")
+	}
+	if rsp.MaxSessionDurationMinutes == 0 {
+		rsp.MaxSessionDurationMinutes = 60
+	}
+	if rsp.Path == "" {
+		rsp.Path = "/"
+	}
+}
+
+func (rsp *IAMRoleStackProps) validateProps() error {
+	var errs []error
+	if rsp.MaxSessionDurationMinutes < 60 {
+		errs = append(errs, fmt.Errorf("maxSessionDurationMinutes must be at least 60"))
+	}
+	if !strings.HasPrefix(rsp.Path, "/") {
+		errs = append(errs, fmt.Errorf("path must start with a /"))
+	}
+	if len(rsp.Policy) == 0 {
+		errs = append(errs, fmt.Errorf("policy cannot be empty"))
+	}
+	if _, err := arn.Parse(rsp.TrustedArn); err != nil {
+		errs = append(errs, fmt.Errorf("failed to parse trustedArn: %w", err))
+	}
+	return errors.Join(errs...)
+}
+
+func newIAMRoleStack(scope constructs.Construct, id string, props *IAMRoleStackProps) (awscdk.Stack, error) {
+	var sprops awscdk.StackProps
+	if props != nil {
+		sprops = props.StackProps
+	}
+	stack := awscdk.NewStack(scope, &id, &sprops)
+
+	roleProps := &awsiam.RoleProps{
+		AssumedBy:          awsiam.NewArnPrincipal(jsii.String(props.TrustedArn)),
+		Description:        jsii.String("Acorn created IAM Role"),
```

Comment:
Could we make this configurable as well and default it to the current description you have here? I envision a user wanting to set something for this.

```
acorn run ghcr.io/acorn-io/aws/iam/role:v0.1.0 --description "foobarbaz"
```

Reply:
Done!